addon/mixins/scoped-query-cache.js

```diff
@@ -73,5 +76,29 @@ export default Ember.Mixin.create({
 
       return results;
     });
+  },
+
+  scheduleCacheDecay(delay = CACHE_DECAY_TIMEOUT, callback) {
+    this.decayId = Ember.run.later(() => {
+      if (this.isDestroying || this.isDestroyed) {
+        return;
+      }
+
+      this.cache = this._createCache();
+
+      callback && typeof callback === 'function' && callback();
+    }, delay);
+  },
+
+  cancelCacheDecay() {
```

Comment:
This likely needs to be called in `willDestroy` as you may have the `later` scheduled.